pkg/controller/sparkapplication/batchscheduler/factory.go

```diff
@@ -0,0 +1,38 @@
+package batchscheduler
+
+import (
+	"sync"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/controller/sparkapplication/batchscheduler/interface"
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/controller/sparkapplication/batchscheduler/volcano"
+)
+
+type schedulerInitializeFunc func(config *rest.Config) schedulerinterface.BatchScheduler
+
+var manageMutex sync.Mutex
```

Comment:
Put both into the same var block to tell that the map is guarded by the mutex, e.g.,:

```
var (
  mutex sync.Mutex
  schedulers map[string]schedulerInitializeFunc
)
```